app/controllers/exports_controller.rb

```diff
@@ -72,10 +72,30 @@ def include_parameters
   end
 
   def set_filename
-    filename = export.csv
-    filename += "_#{@labware.human_barcode}" if export.filename['include_barcode']
+    # The filename falls back to the csv template attribute if no filename is provided.
+    filename = export.filename['name'] || export.csv
+    filename = build_filename(filename)
+    file_extension = export.file_extension || 'csv'
+    response.headers['Content-Disposition'] = "attachment; filename=\"#{filename}.#{file_extension}\""
+  end
+
+  def build_filename(filename)
+    # Append or prepend the give barcodes to the filename if specified in the export configuration.
+    filename = handle_filename_barcode(filename, @labware, export.filename['labware_barcode'])
+    filename = handle_filename_barcode(filename, @labware.parents&.first, export.filename['parent_labware_barcode'])
+
+    # Append the page number to the filename if specified in the export configuration.
     filename += "_#{@page + 1}" if export.filename['include_page']
-    response.headers['Content-Disposition'] = "attachment; filename=\"#{filename}.csv\""
+    filename
+  end
+
+  def handle_filename_barcode(filename, labware, options)
+    return filename if options.blank? || labware.blank?
+
+    barcode = labware.barcode.human
```

Comment:
Interested to see `labware.barcode.human` here, as @StephenHulme found recently that the `human` attribute wasn't present - see discussion here https://github.com/sanger/limber/pull/1746#discussion_r1657294204

Perhaps it's because Stephen was working with tubes and you're working with plates - maybe the API returns different things. Wonder if it would be worth doing something like 
```
 barcode = labware.barcode.human || labware.barcode.machine
```
in case human is blank? Although in that case, not sure if `labware.barcode.human` would return `nil` or blow up.

Reply:
Good point, the reason I did this was because labware.human_barcode didnt exist on some of my tests but labware.barcode.human did. I haven't run into a nil case for it yet but happy to add some handling for it. Would a machine barcode be more useful than just showing nil? They can manually edit it if it there are any issues.

Reply:
I guess from the users point of view the aim is to match the export file with the physical plate using it's barcode label, so unless what's in the filename matches what's displayed on the plate label (human readable) it is not useful. Important for driver files where they are loading a plate on the robot and using the file to drive transfers for it.
And they (mostly) don't transfer to/from tubes on robots.
Odd there is no human attribute on tube labware barcodes. Maybe we should fix that.